apps/src/StudioApp.js

```diff
@@ -285,14 +286,20 @@ StudioApp.prototype.init = function(config) {
     dom.addClickTouchEvent(showCode, _.bind(function() {
       if (this.editCode) {
         var result;
+        var nonDropletError = false;
+        // are we trying to toggle from blocks to text (or the opposite)
+        var fromBlocks = this.editor.currentlyUsingBlocks;
         try {
           result = this.editor.toggleBlocks();
         } catch (err) {
+          nonDropletError = true;
           result = {error: err};
         }
         if (result && result.error) {
-          // TODO (cpirich) We could extract error.loc to determine where the
-          // error occurred and highlight that error
+          newrelic.addPageAction(newrelic.PageAction.BlockToCodeError, {
```

Comment:
I'd vote for calling this something like `DropletTransitionError`, so we don't forget that it also includes "code to block" errors